server/src/test/java/io/druid/server/coordinator/DruidCoordinatorBalancerTester.java

```diff
@@ -46,7 +46,6 @@ protected void moveSegment(
 
     if (!toPeon.getSegmentsToLoad().contains(segmentToMove) &&
         !currentlyMovingSegments.get("normal").containsKey(segmentName) &&
-        !toServer.getSegments().containsKey(segmentName) &&
```

Comment:
Because toServer's segments are not updated in the block of this if statement (unlike in production code, from where this condition was apparently copied), and this expression is either always true or false. It wasn't noticed before because due to the other bug, fixed a few lines below, all tests using `DruidCoordinatorBalancerTester` basically always passed.